src/App.tsx

```diff
@@ -1,61 +1,120 @@
+/* eslint-disable @typescript-eslint/indent */
+import { useState } from 'react';
+
 import './App.scss';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { ToDo } from './types/todo';
+import { findUserById, getTodosMaxId } from './service';
+
+export const App: React.FC = () => {
+  const tasks = todosFromServer.map(todo => {
+    return {
+      ...todo,
+      user: findUserById(todo.userId),
+    };
+  });
+
+  const [todos, setTodos] = useState<ToDo[]>([...tasks]);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+    setErrorTitle(false);
+    setErrorUser(false);
+  };
+
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = event => {
+    event.preventDefault();
+
+    setErrorTitle(!title);
+    setErrorUser(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    setTodos([
+      ...todos,
+      {
+        id: getTodosMaxId(todos) + 1,
+        title: title,
+        completed: false,
+        userId: Number(userId),
+        user: findUserById(Number(userId)),
+      },
+    ]);
+
+    resetForm();
+  };
+
+  const handleUserIdChange: React.ChangeEventHandler<
+    HTMLSelectElement
+  > = ev => {
+    setUserId(+ev.target.value);
+    setErrorUser(false);
+  };
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+  const handleTitleChange: React.ChangeEventHandler<HTMLInputElement> = ev => {
+    setTitle(ev.target.value);
+    setErrorTitle(false);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
+          <input
+            id="title"
+            value={title}
+            placeholder="Enter a title"
+            type="text"
+            data-cy="titleInput"
+            onChange={handleTitleChange}
+          />
+          {errorTitle && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="title">User: </label>
```

Comment:
The `label` for the user select input has an incorrect `htmlFor` attribute value. It should be `htmlFor="select"` to match the `id` of the select element.